### PR TITLE
Update to latest version of clap

### PR DESCRIPTION
**Pull request by flavio**
I did a quick smoke test, plus I ran the e2e tests.

Everything seems to be working.

### CODE REVIEW

**Comment by flavio** [comment]
Supersedes https://github.com/kubewarden/kwctl/pull/117

File: src/cli.rs [hunk 1]
```diff
@@ -34,352 +32,351 @@ lazy_static! {
...
-pub fn build_cli() -> clap::App<'static, 'static> {
+pub fn build_cli() -> clap::App<'static> {
     App::new(crate_name!())
         .version(crate_version!())
         .author(crate_authors!())
         .about(crate_description!())
-        .arg(Arg::with_name("verbose").short("v").help("Increase verbosity"))
+        .arg(Arg::new("verbose").short('v').help("Increase verbosity"))
         .subcommand(
-            SubCommand::with_name("policies")
```
Comment (ereslibre) [inline src/cli.rs:44]: I was a bit suprised by this API changed on clap's side (SubCommand -> Into<App>). Seems that everything is working fine. I tried some commands, also with the generated `--help` and such and everything seems to work fine. e2e tests are passing as well.

**Review by ereslibre — APPROVED** [review]

**Comment by ereslibre** [comment]
I'd go ahead and merge this PR.

**Comment by flavio** [comment]
rebased against main

**Pull request merged**